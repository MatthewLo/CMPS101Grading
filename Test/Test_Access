https://github.com/ShobhitMaheshwari/CMPS101Grading/blob/master/Test/Test_Access.c
5
Check list access functions
gcc -std=c99 -Wall -c Test_Access.c > w 2>&1
gcc -std=c99 -Wall -o Test_Access Test_Access.o ../List.o &> r
./Test_Access
valgrind --leak-check=full --log-file="memoryLeak" ./Test_Access
