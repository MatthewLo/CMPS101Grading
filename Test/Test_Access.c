
#include<stdio.h>
#include<stdlib.h>
#include "../List.h"
int main(){
	FILE* out;
	out = fopen("Test_Access.out", "w");
	if( out==NULL ){
		printf("Unable to write to file\n");
		exit(EXIT_FAILURE);
	}
	List a = newList();
	append(a, 3);
	printList(out, a); fprintf(out, "\n"); //3
	prepend(a, 4);
	printList(out, a); fprintf(out ,"\n");//4 3
	fprintf(out, "%d\n", length(a));
	append(a, 5);
	printList(out, a); 	fprintf(out, "\n");//4 3 5
	fprintf(out, "%d\n", length(a)); //3
	fprintf(out, "%d\n", front(a)); // 4
	fprintf(out, "%d\n", back(a));//5

	fclose(out);
	freeList(&a);
	return 0;
}
