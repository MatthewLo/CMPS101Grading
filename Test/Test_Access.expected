3
4 3
2
4 3 5
3
4
5
